OpenRA.Mods.Common/Traits/Render/Hovers.cs

```diff
@@ -43,7 +43,9 @@ public IEnumerable<IRenderable> ModifyRender(Actor self, WorldRenderer wr, IEnum
 			if (self.World.Paused || IsTraitDisabled)
 				return r;
 
-			var visualOffset = !aircraft || self.CenterPosition.Z > 0 ? (int)Math.Abs((self.ActorID + Game.LocalTick) / 5 % 4 - 1) - 1 : 0;
+			var visualOffset = !aircraft || self.World.Map.DistanceAboveTerrain(self.CenterPosition).Length > 0
```

Comment:
Actually this trait shouldn't care about aircraft at all. Especially with the PR that only enables it at cruise altitude.

I'd lobby for keeping it `> 0`, but the TS MLRS is will need to hover only when above the ground (not when EMP-ed) (granted, that should still be possible now). So for the sake of both flexibility and clean architecture I suggest adding a `(Min)HoveringAltitude` or similar and removing the dependency on `Aircraft` entirely.